compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5606,13 +5571,86 @@ private BType resolveQueryType(SymbolEnv env, BLangExpression selectExp, BType c
         }
     }
 
-    private BType getQueryTableType(BLangQueryExpr queryExpr, BType constraintType) {
+    void solveSelectTypeAndResolveType(BLangQueryExpr queryExpr, BLangExpression selectExp, BType type,
+                                       BType collectionType, List<BType> selectTypes, List<BType> resolvedTypes,
+                                       SymbolEnv env, AnalyzerData data, boolean isReadonly) {
+        BType selectType, resolvedType;
+        type = Types.getReferredType(type);
+        switch (type.tag) {
+            case TypeTags.ARRAY:
+                BType elementType = ((BArrayType) type).eType;
+                if (elementType.tag == TypeTags.INTERSECTION) {
```

Comment:
Why do we need to consider intersections here for array type?

Reply:
My bad. I forgot to remove this. Thanks.